sdk/containerregistry/azure-containers-containerregistry-perf/src/main/java/com/azure/containers/containerregistry/perf/DownloadBlobTests.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.containers.containerregistry.perf;
+
+import com.azure.containers.containerregistry.perf.core.ServiceTest;
+import com.azure.core.util.Context;
+import com.azure.perf.test.core.PerfStressOptions;
+import reactor.core.publisher.Mono;
+
+import java.io.OutputStream;
+import java.nio.channels.Channels;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.containers.containerregistry.perf.core.TestInputStream.generateAsyncStream;
+import static com.azure.containers.containerregistry.perf.core.Utils.REPOSITORY_NAME;
+
+public class DownloadBlobTests extends ServiceTest<PerfStressOptions> {
+    private volatile AtomicReference<String> digest = new AtomicReference<>();
```

Comment:
Given this is an `AtomicReference` does it need to be volatile as well? Also, is there an exact reason we need this? Or, since this doesn't change during the runtime of the performance test could this be turned into `String[]` where there is only one index and it's the digest value, this way we don't add the locking/atomic overhead of volatile and however `AtomicReference` manages to make the reference atomic.

Reply:
changed to array, no need in sync.